shared/gateway/gateway.go

```diff
@@ -38,53 +42,131 @@ type Gateway struct {
 	maxCallRecvMsgSize      uint64
 }
 
-// Start the gateway service. This serves the HTTP JSON traffic on the specified
-// port.
+// New returns a new gateway server which translates HTTP into gRPC.
+// Accepts a context and optional http.ServeMux.
+func NewValidator(
+	ctx context.Context,
+	callerId string,
```

Comment:
The `callerId` parameter doesn't make sense because `NewValidator` will always be called with `validator-rpc`. You can use the literal string value in the `&Gateway` expression.

Reply:
The challenge is not the New function, but the overridden SharedRegistry  Start, Status, and Stop funcs. While the latter two can be merged without harming the original beacon and Validator nodes, the former (Start) has diff code forks .